chains/runtime-common/src/processors/v2/symbiotic_message_processor.rs

```diff
@@ -0,0 +1,219 @@
+                validators,
+                external_index,
+            )
+            .map_err(|error| MessageProcessorError::ProcessMessage(error))?;
+            Ok(())
+        }
+    }
+}
+
+#[derive(MessageProcessor)]
+pub struct SymbioticMessageProcessor<
+    T,
+    GatewayAddress,
+    DefaultClaimer,
+    EthereumNetwork,
+    EthereumUniversalLocation,
+    TanssiUniversalLocation,
+    XcmProcessor,
+    XcmWeigher,
+>(
+    PhantomData<(
+        T,
+        GatewayAddress,
+        DefaultClaimer,
+        EthereumNetwork,
+        EthereumUniversalLocation,
+        TanssiUniversalLocation,
+        XcmProcessor,
+        XcmWeigher,
+    )>,
+);
+
+impl<
+        T,
+        AccountId,
+        GatewayAddress,
+        DefaultClaimer,
+        EthereumNetwork,
+        EthereumUniversalLocation,
+        TanssiUniversalLocation,
+        XcmProcessor,
+        XcmWeigher,
+    > MessageProcessorWithFallback<AccountId>
+    for SymbioticMessageProcessor<
+        T,
+        GatewayAddress,
+        DefaultClaimer,
+        EthereumNetwork,
+        EthereumUniversalLocation,
+        TanssiUniversalLocation,
+        XcmProcessor,
+        XcmWeigher,
+    >
+where
+    T: snowbridge_pallet_inbound_queue::Config
+        + pallet_xcm::Config
+        + snowbridge_pallet_system::Config
+        + pallet_external_validators::Config,
+    [u8; 32]: From<<T as frame_system::Config>::AccountId>,
+    GatewayAddress: Get<H160>,
+    DefaultClaimer: Get<<T as frame_system::Config>::AccountId>,
+    EthereumNetwork: Get<NetworkId>,
+    EthereumUniversalLocation: Get<InteriorLocation>,
+    TanssiUniversalLocation: Get<InteriorLocation>,
+    XcmProcessor: ExecuteXcm<<T as pallet_xcm::Config>::RuntimeCall>,
+    XcmWeigher: WeightBounds<<T as pallet_xcm::Config>::RuntimeCall>,
+{
+    type Fallback = SymbioticFallbackProcessor<
+        T,
+        AssetTrapFallbackProcessor<
+            T,
+            GatewayAddress,
+            DefaultClaimer,
+            EthereumNetwork,
+            EthereumUniversalLocation,
+            TanssiUniversalLocation,
+            XcmProcessor,
+            XcmWeigher,
+        >,
+        GatewayAddress,
+        DefaultClaimer,
+        EthereumNetwork,
+        EthereumUniversalLocation,
+        TanssiUniversalLocation,
+        XcmProcessor,
+        XcmWeigher,
+    >;
+    type ExtractedMessage = SymbioticMessage<T>;
+
+    fn try_extract_message(
+        _sender: &AccountId,
+        message: &Message,
+    ) -> Result<Self::ExtractedMessage, MessageExtractionError> {
+        let gateway_proxy_address = T::GatewayAddress::get();
+        try_extract_message(message, gateway_proxy_address)
+    }
+
+    fn process_extracted_message(
+        _sender: AccountId,
+        extracted_message: Self::ExtractedMessage,
+    ) -> Result<[u8; 32], MessageProcessorError> {
+        process_message(extracted_message).map(|_| [0; 32])
```

Comment:
Is it expected that we always return `[0; 32]` from `process_extracted_message`? Why is that? If it's not planned to be used, we can change the trait and have this method return `()`.